src/result.ts

```diff
@@ -879,3 +890,43 @@ export const toString = <T, E>(result: Result<T, E>): string => {
   const body = (isOk(result) ? unwrap(result) : unwrapErr(result)).toString();
   return `${result.variant}(${body})`;
 };
+
+/** A lightweight object defining how to handle each variant of a Maybe. */
+export type Matcher<T, E, A> = {
+  Ok: (value: T) => A;
+  Err: (error: E) => A;
+};
+
+/**
+  Performs the same basic functionality as `getOrElse`, but instead of simply
+  unwrapping the value if it is `Ok` and applying a value to generate the same
+  default type if it is `Nothing`, lets you supply functions which may transform
+  the wrapped type if it is `Ok` or get a default value for `Nothing`.
+
+  This is kind of like a poor-man's version of pattern matching, which
+  JavaScript currently lacks.
+
+  ```ts
+  import { match, ok, err } from 'true-myth/result';
+
+  const double = (n: number) => n * 2;
+  const length = (s: string) => s.length;
+
+  const anOk = ok<number, string>(12);
+  console.log(match({ Ok: double, Err: length }, anOk)); // 24
+
+  const anErr = err<number, string>('oh no');
+  console.log(match({ Ok: double, Err: length }, anErr)); // 0
```

Comment:
Thinking about these, I'm not sure they're the best examples. Matching on `anOk` when you _define_ it from the get-go as an `Ok` doesn't really show the full value of the function.

Reply:
That's a good point. What about something like this?

```ts
import { Result, match } from 'true-myth/result';

const double = (n: number) => n * 2;
const length = (s: string) => s.length;

const double = (aNumberResult: Result<number, string>): number =>
  match({ Ok: double, Err: length }, aNumberResult);
```

Reply:
Yeah, I think that's a lot better because you can't know whether you'll have an `Ok` or an `Err` at the site of matching.